pilot/pkg/config/kube/crd/client.go

```diff
@@ -455,6 +455,10 @@ func (cl *Client) Delete(typ, name, namespace string) error {
 		return fmt.Errorf("missing type %q", typ)
 	}
 
+	if len(namespace) == 0 {
+		namespace = meta_v1.NamespaceDefault
+	}
+
```

Comment:
It's a bug if namespace is empty. I would either return an error _or_ update the semantics of `Delete` to make it handle cluster scoped resources as well. In the latter case, that means omitting `Namespace(namespace)` from the builder list on line 463 if namespace is empty.

Reply:
> In the latter case, that means omitting Namespace(namespace) from the builder list on line 463 if namespace is empty.

Hmm, I meant to do so, but I noticed that we did namespace conversion in Create/Update (ConvertObject), so I thought it might be better to keep consistency here.
Anyway, both of the ways are fine to me.

Reply:
Done

Reply:
It's weird, tests are passing locally, but in CI, it always fail with the error:
```
--- FAIL: TestTempWorkspace (16.99s)
    --- PASS: TestTempWorkspace/StoreInvariant (4.40s)
    --- FAIL: TestTempWorkspace/istioConfig (0.20s)
    	config.go:464: Error deleting config rbac-config//default, err: Namespace parameter required.
```